src/promptflow/promptflow/_sdk/data/executable/main_csharp.py.jinja2

```diff
@@ -0,0 +1,140 @@
+        st.session_state.messages.append(("user", kwargs))
+        session_state_history = dict()
+        session_state_history.update({"inputs": kwargs})
+        with container:
+            render_message("user", kwargs)
+        # Force append chat history to kwargs
+{% if is_chat_flow %}
+{{ ' ' * indent_level * 2 }}response = run_flow({'{{chat_history_input_name}}': get_chat_history_from_session(), **kwargs})
+{% else %}
+{{ ' ' * indent_level * 2 }}response = run_flow(kwargs)
+{% endif %}
+        st.session_state.messages.append(("assistant", response))
+        session_state_history.update({"outputs": response})
+        st.session_state.history.append(session_state_history)
+        with container:
+            render_message("assistant", response)
+
+
+    def run_flow(data: dict) -> dict:
+{% if flow_path %}
+{{ ' ' * indent_level * 2 }}flow = Path('{{flow_path}}')
+{{ ' ' * indent_level * 2 }}dump_path = Path('{{flow_path}}').parent
+{% else %}
+{{ ' ' * indent_level * 2 }}flow = Path(__file__).parent / "flow"
+{{ ' ' * indent_level * 2 }}dump_path = flow.parent
+{% endif %}
+        if flow.is_dir():
+            os.chdir(flow)
+        else:
+            os.chdir(flow.parent)
+        invoker = load_flow(flow)
+        result = invoker.invoke(data)
+        return result
+
+    image = Image.open(Path(__file__).parent / "logo.png")
+    st.set_page_config(
+        layout="wide",
+        page_title="{{flow_name}} - Promptflow App",
+        page_icon=image,
+        menu_items={
+            'About': """
+            # This is a Promptflow App.
+
+            You can refer to [promptflow](https://github.com/microsoft/promptflow) for more information.
+            """
+        }
+    )
+    # Set primary button color here since button color of the same form need to be identical in streamlit, but we only need Run/Chat button to be blue.
+    st.config.set_option("theme.primaryColor", "#0F6CBD")
+    st.title("{{flow_name}}")
+    st.divider()
+    st.chat_message("assistant").write("Hello, please input following flow inputs.")
+    container = st.container()
+    with container:
+        show_conversation()
+
+    with st.form(key='input_form', clear_on_submit=True):
+        settings_path = os.path.join(os.path.dirname(__file__), "settings.json")
+        if os.path.exists(settings_path):
+            with open(settings_path, "r", encoding="utf-8") as file:
+                json_data = json.load(file)
+            environment_variables = list(json_data.keys())
+            for environment_variable in environment_variables:
+                secret_input = st.sidebar.text_input(label=environment_variable, type="password", placeholder=f"Please input {environment_variable} here. If you input before, you can leave it blank.")
+                if secret_input != "":
+                    os.environ[environment_variable] = secret_input
+
+{% for flow_input, (default_value, value_type) in flow_inputs.items() %}
```

Comment:
what's the difference with this new jinja2 and exsiting ones?

Reply:
Csharp jinja call `flow function call` invoke function (only support csharp flow for now in my pr),  which I think Han also will provide a pr to add this function according to offline sync with Xingzhi. 

Since it's not fully ready, I don't want it break executable.

Reply:
As for test_flow_main.jinja, it's only for chat flow with one input and one output, which it's not ready in Csharp flow